docs/pages/2020_Intro_Week/sections/basics.rst

```diff
@@ -3,3 +3,961 @@
+
+Pseudopotentials
+~~~~~~~~~~~~~~~~
+
+From the graph you generated in  section :ref:`2020_virtual_aiidagraph`,
+find the ``PK`` of the pseudopotential file (LDA). Load it and
+show what elements it corresponds to by typing:
+
+.. code:: python
+
+    upf = load_node(PK)
+    upf.element
+
+All methods of ``UpfData`` are accessible by typing ``upf.`` and then pressing ``TAB``.
+
+k-points
+~~~~~~~~
+
+A set of k-points in the Brillouin zone is represented by an instance of the
+``KpointsData`` class. Choose one from the graph of
+produced in section :ref:`2020_virtual_aiidagraph`,
+load it as ``kpoints`` and inspect its content:
+
+.. code:: python
+
+    kpoints.get_kpoints_mesh()
+
+Then get the full (explicit) list of k-points belonging to this mesh using
+
+.. code:: python
+
+    kpoints.get_kpoints_mesh(print_list=True)
+
+If this throws an ``AttributeError``, it means that the kpoints instance does not represent a regular mesh but rather a list of k-points defined by their crystal coordinates (typically used when plotting a band structure).
+In this case, get the list of k-points coordinates using
+
+.. code:: python
+
+    kpoints.get_kpoints()
+
+Conversely, if the `KpointsData` node `does` actually represent a mesh, this method is the one, that when called, will throw an ``AttributeError``.
+
+If you prefer Cartesian (rather than crystal) coordinates, type
+
+.. code:: python
+
+    kpoints.get_kpoints(cartesian=True)
+
+For later use in this tutorial, let us try now to create a kpoints instance,
+to describe a regular (2 x 2 x 2) mesh of k-points, centered at the Gamma
+point (i.e. without offset). This can be done with the following commands:
+
+.. code:: python
+
+    KpointsData = DataFactory('array.kpoints')
+    kpoints = KpointsData()
+    kpoints_mesh = 2
+    kpoints.set_kpoints_mesh([kpoints_mesh] * 3)
+    kpoints.store()
+
+This function loads the appropriate class defined in a string (here
+``array.kpoints``). Therefore, ``KpointsData`` is not a class instance, but
+the kpoints class itself!
+
+While it is also possible to import ``KpointsData`` directly, it is recommended
+to use the ``DataFactory`` function instead, as this is more future-proof:
+even if the import path of the class changes in the future, its entry point
+string (``array.kpoints``) will remain stable.
+
+Parameters
+~~~~~~~~~~
+
+Dictionaries with various parameters are represented in AiiDA by ``Dict`` nodes.
+Get the PK and load the input parameters of a calculation in the graph produced in  section :ref:`2020_virtual_aiidagraph`.
+Then display its content by typing
+
+.. code:: python
+
+    params = load_node('<IDENTIFIER>')
+    YOUR_DICT = params.get_dict()
+    YOUR_DICT
+
+Modify the python dictionary ``YOUR_DICT`` so that the wave-function cutoff is now set to 20
+Ry. Note that you cannot modify an object already stored in the database. To
+write the modified dictionary to the database, create a new object of class ``Dict``:
```

Comment:
```suggestion
Modify the python dictionary ``YOUR_DICT`` so that the wave-function cutoff is now set to 20 Ry.
Note that you cannot modify an object already stored in the database.
To write the modified dictionary to the database, create a new object of class ``Dict``:
```